Feat/add test case for get boosts

bhavyagosai opened the pull request:
<!--- Please provide a general summary of your changes in the title above -->

# Pull Request type

<!-- Please try to limit your pull request to one type; submit multiple pull requests if needed. -->

Please add the labels corresponding to the type of changes your PR introduces:

- Testing

Resolves: #603 



## Other information

my test covers the lines `53-61` ie `getBoosts()` for the file `apiService.ts`

<img width="1147" alt="image" src="https://github.com/starknet-id/starknet.quest/assets/64588227/f406f558-893f-4043-9b71-22dd14295dca">

bhavyagosai committed:
feat: add test cases for getBoosts api

bhavyagosai committed:
feat: added response type for getBoosts

ayushtom commented on app/quest-boost/page.tsx:
```diff
@@ -22,7 +22,7 @@ export default function Page() {
     try {
       const res = await getBoosts();
-      setBoosts(res);
+      if (res) setBoosts(res);
```
can set this as 
```
if(!res) {
// show error message on screen saying - No boosts available now
(you can use error message screen component in the codebase to do this)
}
setBoosts(res);
```

bhavyagosai replied:
@ayushtom This is the current error message (No quests are being boosted at the moment)

<img width="1430" alt="image" src="https://github.com/starknet-id/starknet.quest/assets/64588227/d2b85aab-0aed-46a8-9b8f-e61a275173c1">

Should I change it to the one below? But then the ErrorScreen component takes the whole page height and there is content below it...

<img width="1431" alt="image" src="https://github.com/starknet-id/starknet.quest/assets/64588227/8cdf644a-5977-4730-b603-5c9cdc733cb9">

ayushtom replied:
Hmm , actually what you can do is go with the first option, but also keep the same space in the section as the cards were taking and place the text in the center. Currently it looks a little stuck together vertically. wdyt?

bhavyagosai replied:
makes sense, I was thinking something of the same sort too

bhavyagosai committed:
Merge remote-tracking branch 'origin' into feat/add-test-case-for-getBoosts

bhavyagosai committed:
fix: improve ui + handle edge cases

bhavyagosai commented:
- Have added checks wherever `boost/get_boosts` api was being fetched and displayed
- Found one component `QuestCategories` where the api was being used but not sure if currently the component is being used anywhere as I found it only to be declared but never used. Nevertheless, have updated it as well.

Something I wanted to point out was that, currently there are no loaders displayed while any loading occurs while fetching the data through apis. I feel we ought to add a placeholder/loader wherever data is loading.

ayushtom committed:
Merge branch 'testnet' into feat/add-test-case-for-getBoosts

ayushtom commented:
<img width="1452" alt="Screenshot 2024-04-27 at 2 16 27 PM" src="https://github.com/starknet-id/starknet.quest/assets/41674634/483137e4-1462-4a1d-9bf5-4c46ab0ed932">

Hey i cant see the boosts when i come to this page. Also your suggestion regarding loaders is right. Can you add a loader on this page when we fetch boosts? I think currently the - "No boosts available" text comes up first when i come on this page

bhavyagosai commented:
> I think currently the - "No boosts available" text comes up first when i come on this page

Yeah exactly. I'll fix them

bhavyagosai committed:
feat: made loadingScreen more modular

bhavyagosai committed:
feat: added proper loading states + code imp

bhavyagosai committed:
feat: improved code logic for BoostCard

bhavyagosai committed:
Merge remote-tracking branch 'origin' into feat/add-test-case-for-getBoosts

bhavyagosai commented:
@ayushtom idk about not being able to see the boosts, I am able to see them. You can confirm this again now I have updated some things.

- added loader in `\quest-boost` page.
- moved the api call functions inside useEffect() as they were being used only inside that scope.
- improved code logic for BoostCard useEffect() check for any/all quests being completed by user

One more thing:
https://github.com/bhavyagosai/starknet.quest/blob/e54f9a762491e468510feb405fe401be60e5f3c5/app/quest-boost/page.tsx#L64 I felt as both boosts and completedQuests array were fetched and displayed here (or used for any logic), so rendering this after they have been fetched. You can correct me if I am wrong here.

Note: have not seen all part of code but maybe small small code improvements can be made elsewhere regarding adding loaders and all 😅

bhavyagosai committed:
temp remove for merging

bhavyagosai committed:
Merge remote-tracking branch 'origin' into feat/add-test-case-for-getBoosts

bhavyagosai committed:
add getBoosts api back

bhavyagosai committed:
format code

ayushtom committed:
Merge branch 'testnet' into feat/add-test-case-for-getBoosts

bhavyagosai committed:
feat: updated loading state with skeleton loading

bhavyagosai committed:
feat: added fallback message if filtered data is empty

bhavyagosai committed:
feat: display both expired and non expired quest boosts

bhavyagosai committed:
Merge remote-tracking branch 'origin' into feat/add-test-case-for-getBoosts

bhavyagosai committed:
format apiService.test.js

ayushtom committed:
Merge branch 'testnet' into feat/add-test-case-for-getBoosts

ayushtom reviewed (approved):
lgtm
